sql/core/src/test/scala/org/apache/spark/sql/execution/systemcatalog/InformationSchemaSuite.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.systemcatalog
+
+import org.apache.spark.sql.{QueryTest, Row}
+import org.apache.spark.sql.test.SharedSQLContext
+
+/**
+ * Information Schema Suite
+ */
+class InformationSchemaSuite extends QueryTest with SharedSQLContext {
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    InformationSchema.build(spark)
+    sql("CREATE TABLE s(a int, b double)")
+    sql("CREATE TABLE t(c int, d double)")
+  }
+
+  override def afterAll(): Unit = {
+    try {
+      sql("DROP TABLE IF EXISTS t")
+      sql("DROP TABLE IF EXISTS s")
+    } finally {
+      super.afterAll()
+    }
+  }
+
+  test("databases/schemata scan") {
+    checkAnswer(
+      sql("select * from databases"),
+      Row("default", "default") :: Row("default", "information_schema") :: Nil)
+
+    checkAnswer(
+      sql("select * from schemata"),
+      Row("default", "default") :: Row("default", "information_schema") :: Nil)
+  }
+
+  test("databases/schemata project") {
+    checkAnswer(
+      sql("select SCHEMA_NAME from databases"),
+      Row("default") :: Row("information_schema") :: Nil)
+
+    checkAnswer(
+      sql("select SCHEMA_NAME from schemata"),
+      Row("default") :: Row("information_schema") :: Nil)
+  }
+
+  test("tables scan") {
+    checkAnswer(
+      sql("select * from tables"),
+      Row("default", "default", "columns", "VIEW") ::
```

Comment:
mhm the indentation is really weird - i don't think we need to indent each line with one more level ...